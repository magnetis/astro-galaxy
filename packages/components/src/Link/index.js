import styled from 'styled-components';
import { fontSize, typography } from 'styled-system';
import { font } from '@magnetis/astro-galaxy-core';

const Link = styled.a`
  color: ${props => props.theme.colors.uranus700};
  text-decoration: underline;
  word-wrap: break-word;
  cursor: pointer;

  &:hover {
    color: ${props => props.theme.colors.uranus500};
  }

  &:focus {
    outline: none;
    box-shadow: 0 0 0 3px rgba(21, 156, 228, 0.4);
  }

  ${font}
  ${fontSize};
`;

Link.displayName = 'Link';

Link.defaultProps = {
  font: 'primary',
  fontSize: 'medium',
};

Link.propTypes = {
  ...fontSize.propTypes,
  ...typography.propTypes,
};

export default Link;
