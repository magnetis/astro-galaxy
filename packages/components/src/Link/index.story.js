import { storiesOf } from '@storybook/react';
import React from 'react';
import Link from './index.js';

storiesOf('buttons|links', module).add('default', () => (
  <>
    <Link>primary font - medium size</Link>
    <br />
    <Link font="secondary" fontSize="small">
      secondary font - small size
    </Link>
  </>
));
