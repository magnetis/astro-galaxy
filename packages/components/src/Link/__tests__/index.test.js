import React from 'react';
import Link from '../index';

describe('Link', () => {
  describe('default render', () => {
    it('should render', () => {
      const json = rendererCreateWithTheme(<Link />).toJSON();

      expect(json).toMatchSnapshot();
    });
  });

  describe('when font is secondary', () => {
    it('should render', () => {
      const json = rendererCreateWithTheme(<Link font="secondary" />).toJSON();

      expect(json).toMatchSnapshot();
    });
  });

  describe('when size is small', () => {
    it('should render', () => {
      const json = rendererCreateWithTheme(<Link fontSize="small" />).toJSON();

      expect(json).toMatchSnapshot();
    });
  });
});
