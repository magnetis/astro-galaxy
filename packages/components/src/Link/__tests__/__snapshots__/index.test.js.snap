// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link default render should render 1`] = `
.c0 {
  color: #1663a0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  word-wrap: break-word;
  cursor: pointer;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  font-size: medium;
}

.c0:hover {
  color: #159ce4;
}

.c0:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

@media (prefers-reduced-motion:reduce) {

}

<a
  className="c0"
  fontSize="medium"
/>
`;

exports[`Link when font is secondary should render 1`] = `
.c0 {
  color: #1663a0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  word-wrap: break-word;
  cursor: pointer;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  font-size: medium;
}

.c0:hover {
  color: #159ce4;
}

.c0:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

@media (prefers-reduced-motion:reduce) {

}

<a
  className="c0"
  fontSize="medium"
/>
`;

exports[`Link when size is small should render 1`] = `
.c0 {
  color: #1663a0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  word-wrap: break-word;
  cursor: pointer;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  font-size: small;
}

.c0:hover {
  color: #159ce4;
}

.c0:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(21,156,228,0.4);
}

@media (prefers-reduced-motion:reduce) {

}

<a
  className="c0"
  fontSize="small"
/>
`;
